arch/arm/core/Kconfig

```diff
@@ -24,6 +24,7 @@ config CPU_CORTEX_M
 	select ARCH_HAS_SUSPEND_TO_RAM
 	select ARCH_HAS_CODE_DATA_RELOCATION
 	select ARCH_SUPPORTS_ROM_START
+	select ARCH_HAS_LLEXT_SUPPORT if !SOC_SERIES_M48X
```

Comment:
This doesn't seem the place to put this, should be selected by SOC Kconfig files if they support it

Reply:
Can you please explain further? Dynamic loading needs to be adapted to the specific CPU architecture, but should work seamlessly on different SOCs. The single exception above is due to #63167 and will be removed fixing that issue.
Also, with a per-SOC rework, this single line will become 152 separate `select`s for Cortex-M alone...

Reply:
Currently there is no Kconfig limit, so it could be enabled even on a MIPS target (and it will fail to build, let alone work). I wanted to be able to design better tests, and therefore I needed to limit them to at least the configurations that _should_ work.
Perhaps you would you prefer to disable this globally and only add the configurations that have been tested in HW? (I didn't go that way because I think it would limit user feedback a lot, at a time where llext is still under active development.)

Reply:
It should be enabled on a soc-by-soc basis for socs that support it. There is no way you can guarantee every ARM architecture in the world can support this, so claiming it does is wrong